src/proj_geodesic.jl

```diff
@@ -0,0 +1,152 @@
+immutable Cdouble6
+    x1::Cdouble
+    x2::Cdouble
+    x3::Cdouble
+    x4::Cdouble
+    x5::Cdouble
+    x6::Cdouble
+end
+
+immutable Cdouble15
+    x1::Cdouble
+    x2::Cdouble
+    x3::Cdouble
+    x4::Cdouble
+    x5::Cdouble
+    x6::Cdouble
+    x7::Cdouble
+    x8::Cdouble
+    x9::Cdouble
+    x10::Cdouble
+    x11::Cdouble
+    x12::Cdouble
+    x13::Cdouble
+    x14::Cdouble
+    x15::Cdouble
+end
+
+immutable Cdouble21
+    x1::Cdouble
+    x2::Cdouble
+    x3::Cdouble
+    x4::Cdouble
+    x5::Cdouble
+    x6::Cdouble
+    x7::Cdouble
+    x8::Cdouble
+    x9::Cdouble
+    x10::Cdouble
+    x11::Cdouble
+    x12::Cdouble
+    x13::Cdouble
+    x14::Cdouble
+    x15::Cdouble
+    x16::Cdouble
+    x17::Cdouble
+    x18::Cdouble
+    x19::Cdouble
+    x20::Cdouble
+    x21::Cdouble
+end
+
+type geod_geodesic
+    a::Cdouble
+    f::Cdouble
+    f1::Cdouble
+    e2::Cdouble
+    ep2::Cdouble
+    n::Cdouble
+    b::Cdouble
+    c2::Cdouble
+    etol2::Cdouble
+
+    # Arrays of parameters must be expanded manually,
+    # currently (either inline, or in an immutable helper-type)
+    # In the future, some of these restrictions may be reduced/eliminated.
+    A3x::Cdouble6
+    C3x::Cdouble15
+    C4x::Cdouble21
+
+    geod_geodesic() = new()
+end
+
+@doc """
+Construct an ellipsoid of revolution with
+    
+    equatorial radius a,
+    flattening f,
+
+Remark, we construct it when constructing a [Projection] using the formulas:
+
+    f = (a − b)/a = 1 − sqrt(1-es)
+    n = (a − b)/(a + b) = f /(2 − f)
+    es = (a^2 − b^2)/a^2 = f(2 − f)
+
+Reference: equations (1)-(3) of 
+    Algorithms for geodesics (arXiv:1109.4448v2 [physics.geo-ph] 28 Mar 2012)
+""" ->
+function _geod_geodesic(a::Cdouble, f::Cdouble)
+    g_ptr = pointer_from_objref(geod_geodesic())
```

Comment:
This seems dangerous, and will likely cause segfaults later.  From the docs:

> pointer_from_objref(object_instance)
>
>  Get the memory address of a Julia object as a Ptr. The existence of the resulting Ptr will not protect the object from garbage collection, so you must ensure that the object remains referenced for the whole time that the Ptr will be used.

The critical thing being that we're not holding a julia reference to the `geod_geodesic` instance (rather, we're returning a bare pointer) so garbage collection will eventually collect the instance and our pointer will be left dangling.